src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -82,6 +87,38 @@ public ReadOnlyAddressBook getAddressBook() {
         return favouritedList;
     }
 
+    /**
+     * Iterate through the list of Persons and return number of persons in
+     * the region given
+     * @param region is the region where buyers have their preference
+     *               for their potential property and sellers have their property in
+     */
+    @Override
+    public int getPersonsBasedOnRegion(String region) {
+        int totalPersons = 0;
+        if (Region.isValidRegion(region)) {
+            for (Person person : getFilteredPersonList()) {
+                if (person.getUserType().isBuyer()) { //if is buyer, check region in preference
+                    Preference preference = person.getPreference().isPresent() ? person.getPreference().get() : null;
+                    if (preference != null && preference.getRegion().toString().equals(region)) {
+                        totalPersons++;
+                    }
+                } else { //if is seller, check region in Property
+                    Set<Property> setOfPropertyValues = person.getProperties();
+                    if (!setOfPropertyValues.isEmpty()) {
+                        Iterator<Property> propertyIterator = setOfPropertyValues.iterator();
+                        while (propertyIterator.hasNext()) {
+                            if (propertyIterator.next().getRegion().toString().equals(region)) {
+                                totalPersons++;
+                            }
```

Comment:
Arrowhead here. 
A person would be counted multiple times for multiple properties in this `region`. Is this intended?
Could use `.stream().anyMatch()`.

Reply:
Yup, this is intended. I want to count the number of properties a seller has in a region:
If seller has properties in {East, West, Central}, then I will increment number of People in East, West and Central by 1.
![image](https://user-images.githubusercontent.com/56862300/159886530-cd29cc9d-4a98-4400-9108-1984b816b80e.png)
![image](https://user-images.githubusercontent.com/56862300/159886572-05d2f183-fe8b-406d-ad7d-eabe55873b24.png)

Reply:
![image](https://user-images.githubusercontent.com/56862300/159888340-ef6526c2-5187-4fc8-8bbb-786dc6a7500b.png)

I am not sure how else to change it to become not an arrowhead (if it still is), please advice @cindyangXQ.

Reply:
These 6 consecutive closing brackets make it look quite arrowhead. I am not sure how to solve it though. Could extract the method into `Person`?

Reply:
You could change some of your ifs to be something like this
```
    if (!Region.isValidRegion(region)) {
        return totalPersons;
    }
    for (Person person : getFilteredPersonList()) {
        if (person.getUserType().isBuyer()) { //if is buyer, check region in preference
            Preference preference = person.getPreference().isPresent() ? person.getPreference().get() : null;
            if (preference != null && preference.getRegion().toString().equals(region)) {
                totalPersons++;
            }
        } else { //if is seller, check region in Property
            Set<Property> setOfPropertyValues = person.getProperties();
            if (setOfPropertyValues.isEmpty()) {
              continue;
            }
            Iterator<Property> propertyIterator = setOfPropertyValues.iterator();
            while (propertyIterator.hasNext()) {
                if (propertyIterator.next().getRegion().toString().equals(region)) {
                    totalPersons++;
                }
            }
        }
    }
```

Reply:
Thanks I have made necessary changes